docs/docs/components/guidelines/README.md

```diff
@@ -552,4 +762,823 @@ end user. Generate and use a GUID for the shared secret value, save it to a
+
+  This is the new component, which is customizable via a user prop named
+  `fileTypes` (as suggested in the [Sensible Props](#sensible-props) section),
+  and extends the base component above to allow the user to freely select the
+  types of files to watch for:
+
+  ```javascript
+  const base = require("../common/new-file");
+
+  module.exports = {
+    ...base,
+    key: "cloud_storage-new-file-of-type",
+    name: "New File of Type",
+    description: "Emits an event whenever a file of a specific type is detected",
+    version: "0.0.1",
+    props: {
+      ...base.props,
+      fileTypes: {
+        type: "string[]",
+        label: "File Types",
+        description: "The types of files to watch for",
+        options: [
+          "application/pdf",
+          "application/zip",
+          "image/jpeg",
+          "image/png",
+          // Other types...
+        ],
+      },
+    },
+    methods: {
+      ...base.methods,
+      isRelevant({ type }) {
+        return this.fileTypes.includes(type);
+      },
+    },
+  };
+  ```
+
+- `cloud_storage/sources/new-pdf-document/new-pdf-document.js`
+
+  This component is a simplified version of the above, which also extends the
+  `common/new-file` module but instead of providing the user with an option to
+  select file types of interest it fixes those types to simply `application/pdf`
+  (the type of PDF documents):
+
+  ```javascript
+  const base = require("../common/new-file");
+
+  module.exports = {
+    ...base,
+    key: "cloud_storage-new-pdf-document",
+    name: "New PDF Document",
+    description: "Emits an event whenever a new PDF document is detected",
+    version: "0.0.1",
+    methods: {
+      ...base.methods,
+      isRelevant({ type }) {
+        return type === "application/pdf";
+      },
+    },
+  };
+  ```
+
+- `cloud_storage/sources/new-file/new-file.js`
+
+  Lastly, the `cloud_storage-new-file` component is refactored so that it reuses
+  the new common logic. The new logic provides a canonical implementation of the
+  new `isRelevant` method:
+
+  ```javascript
+  const base = require("../common/new-file");
+
+  module.exports = {
+    ...base,
+    key: "cloud_storage-new-file",
+    name: "New File",
+    description: "Emits an event when a new file is created in the cloud storage drive",
+    version: "0.0.2",
+    methods: {
+      ...base.methods,
+      isRelevant() {
+        return true;
+      },
+    },
+  };
+  ```
+
+#### Good API Usage
+
+In the previous sections we highlighted the fact that the `.app` modules are the
+ones that contain almost all of the logic regarding the interaction with the
+underlying 3rd party app. For example, the
+[`gitlab.app`](/code/components/gitlab/gitlab.app.js) module defines several
+methods that encapsulate calls to and knowledge about the [Gitlab
+API](https://docs.gitlab.com/ee/api/).
+
+Given that the `.app` modules provide such abstractions around external API's,
```

Comment:
Can you replace instances of "API's" with "APIs"? We generally use the latter in these docs since it's not a possessive noun.